### PR TITLE
feat: upgrade to Olive

### PR DESCRIPTION
**Pull request by jfavellar90**

### CODE REVIEW

**Comment by regisb** [comment]
Can you please attempt to upgrade the various requirements in the Docker image? In particular can we run Ubuntu 22.04?

File: tutornotes/templates/notes/build/notes/Dockerfile [hunk 1]
```diff
@@ -19,8 +19,8 @@ ENV PATH /app/venv/bin:${PATH}
 # https://pypi.org/project/setuptools/
 # https://pypi.org/project/pip/
 # https://pypi.org/project/wheel/
-RUN pip install setuptools==62.1.0 pip==22.0.4 wheel==0.37.1
-RUN pip3 install -r requirements/base.txt
+RUN pip install setuptools==62.1.0 pip==22.2.1 wheel==0.37.1
```
Comment (regisb) [inline tutornotes/templates/notes/build/notes/Dockerfile:22]: Can you also upgrade setuptools? https://pypi.org/project/setuptools/

Comment (jfavellar90) [reply tutornotes/templates/notes/build/notes/Dockerfile]: @regisb done

File: tutornotes/templates/notes/build/notes/Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 FROM docker.io/ubuntu:20.04
```
[the remark this answers is not in the document]
Comment (jfavellar90) [reply tutornotes/templates/notes/build/notes/Dockerfile:1]: @regisb the image in this PR is using 20.04 and works properly in a Tutor local installation. However, I created a different Dockerfile in another branch: https://github.com/overhangio/tutor-notes/blob/dockerfile_v3/tutornotes/templates/notes/build/notes/Dockerfile for 22.04. The image generated from this Dockerfile is 150MB bigger compared with the one with 20.04 (due to the usage of [deadsnakes](https://launchpad.net/~deadsnakes/+archive/ubuntu/ppa) PPA to install python 3.8 I think), but it works properly as well when tested in a local installation. Do you see anything to optimize in the 22.04 Dockerfile?

[the remark this answers is not in the document]
Comment (jfavellar90) [reply tutornotes/templates/notes/build/notes/Dockerfile]: @regisb changed the Dockerfile to use Ubuntu 22.04.

File: setup.py [hunk 3]
```diff
@@ -31,7 +31,7 @@
     packages=find_packages(exclude=["tests*"]),
     include_package_data=True,
     python_requires=">=3.7",
-    install_requires=["tutor>=14.0.0,<15.0.0"],
+    install_requires=["tutor>=15.0.0,<16.0.0"],
```
Comment (mariajgrimaldi) [inline setup.py:34]: maybe it'd easier for future migrations to use just `install_requires=["tutor"],`

File: setup.py [hunk 4]
```diff
@@ -31,7 +31,7 @@
     packages=find_packages(exclude=["tests*"]),
     include_package_data=True,
     python_requires=">=3.7",
```
Comment (mariajgrimaldi) [inline setup.py:33]: why python3.7? :thinking:

Comment (jfavellar90) [reply setup.py:33]: Same version as Tutor: https://github.com/overhangio/tutor/commit/a3d7bb0a0c45f583b19d7d9e6174f6f1c078ef87

File: setup.py (hunk 3 above)
Comment (jfavellar90) [reply setup.py:34]: I think this comes from the change in [codejail](https://github.com/eduNEXT/tutor-contrib-codejail/pull/2). This is the way of pinning the version used across almost all plugins.

File: tutornotes/templates/notes/build/notes/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (regisb) [reply tutornotes/templates/notes/build/notes/Dockerfile]: For the record, 3.7 is the current minimally supported version of Python everywhere, until June 2023 https://endoflife.date/python

[the remark this answers is not in the document]
Comment (regisb) [reply tutornotes/templates/notes/build/notes/Dockerfile]: I realize now that it was kind of stupid to ask to upgrade to Ubuntu 22.04... If you would like to upgrade to 22.04 then it's fine, but feel free to revert your changes.

[the remark this answers is not in the document]
Comment (jfavellar90) [reply tutornotes/templates/notes/build/notes/Dockerfile:1]: Reverted to use Ubuntu 20.04

File: tutornotes/templates/notes/build/notes/Dockerfile [hunk 5]
```diff
@@ -1,5 +1,5 @@
...
-MAINTAINER Overhang.io <contact@overhang.io>
+LABEL maintainer="Overhang.io <contact@overhang.io>"
```
Comment (regisb) [inline tutornotes/templates/notes/build/notes/Dockerfile:2]: You may simply remove this label.

Comment (jfavellar90) [reply tutornotes/templates/notes/build/notes/Dockerfile]: Done :)

**Pull request merged**